backend/app/services/aiod.py

```diff
@@ -0,0 +1,164 @@
+
+from app.config import settings
+from app.helpers import Pagination
+from app.schemas.dataset import Dataset
+from app.schemas.ml_model import MLModel
+
+
+class AssetType(Enum):
+    DATASETS: str = "datasets"
+    ML_MODELS: str = "ml_models"
+    PUBLICATIONS: str = "publications"
+    PLATFORMS: str = "platforms"
+
+
+class AIoDClientWrapper:
+    async_client = None
+
+    def start(self):
+        """Instantiate the client. Call from the FastAPI startup hook."""
+        self.async_client = httpx.AsyncClient(verify=settings.AIOD_API.VERIFY_SSL)
+
+    async def stop(self):
+        """Gracefully shutdown. Call from FastAPI shutdown hook."""
+        await self.async_client.aclose()
+        self.async_client = None
+
+    @property
+    def client(self):
+        assert self.async_client is not None
+        return self.async_client
+
+
+aiod_client_wrapper = AIoDClientWrapper()
+
+
+def get_assets_version(asset_type: AssetType) -> str:
+    match asset_type:
+        case AssetType.DATASETS:
+            return settings.AIOD_API.DATASETS_VERSION
+        case AssetType.ML_MODELS:
+            return settings.AIOD_API.ML_MODELS_VERSION
+        case AssetType.PUBLICATIONS:
+            return settings.AIOD_API.PUBLICATIONS_VERSION
+        case AssetType.PLATFORMS:
+            return settings.AIOD_API.PLATFORMS_VERSION
+
+
+async def get_assets(asset_type: AssetType, pagination: Pagination) -> list:
+    res = await aiod_client_wrapper.client.get(
+        urljoin(
+            base=settings.AIOD_API.BASE_URL,
+            url=Path(asset_type.value, get_assets_version(asset_type)).as_posix(),
+        ),
+        params={"offset": pagination.offset, "limit": pagination.limit},
+    )
+
+    if res.status_code != 200:
+        raise HTTPException(
+            status_code=res.status_code,
+            detail=f"Failed to get {asset_type.value} from AIoD. {res.json()}",
+        )
+
+    return res.json()
+
+
+async def get_asset(asset_type: AssetType, asset_id: int) -> Json:
+    res = await aiod_client_wrapper.client.get(
+        urljoin(
+            base=settings.AIOD_API.BASE_URL,
+            url=Path(
+                asset_type.value, get_assets_version(asset_type), str(asset_id)
+            ).as_posix(),
+        ),
+    )
+
+    if res.status_code != 200:
+        raise HTTPException(
+            status_code=res.status_code,
+            detail=f"Failed to get {asset_type.value} from AIoD. {res.json()}",
+        )
+
+    return res.json()
+
+
+async def get_assets_count(asset_type: AssetType, filter_query: str = None) -> int:
+    if filter_query is None:
+        res = await aiod_client_wrapper.client.get(
+            urljoin(
+                base=settings.AIOD_API.BASE_URL,
+                url=Path(
+                    f"counts/{asset_type.value}", get_assets_version(asset_type)
+                ).as_posix(),
+            ),
+        )
+    else:
+        res = await aiod_client_wrapper.client.get(
```

Comment:
Please, provide a comment on this. Sometimes, comments are a smell, but here it is a small hack enforced by AIoD - the search API allows you to get the number of records that match the search query (+ one record itself ==> limit=1), where the num of records is returned in the "total_hits" field. 

Imo this should be a part of the AIoD /counts API and this is a temp hack.

Reply:
Sure, I will add docstrings for this functions and add a note for this one